test/unit/permutations/test_permutations.cpp

```diff
@@ -204,66 +206,94 @@ void testApplyRowPermutations(SizeType n, SizeType nb) {
   }
 }
 
-// Permute columns or rows in reverse order.
-// Each column or row of the input matrix has its index as a value.
+// Each column/row of the input matrix has its index as a value.
+//
+// The portion of the input matrices rows/columns specified by `i_begin` and `i_end` are placed in
+// reverse order into the output matrix.
 template <Backend B, Device D, class T, Coord C>
-void testPermutations(SizeType n, SizeType nb) {
+void testPermutations(SizeType n, SizeType nb, SizeType i_begin, SizeType i_end) {
   Matrix<SizeType, Device::CPU> perms(LocalElementSize(n, 1), TileElementSize(nb, 1));
   Matrix<T, Device::CPU> mat_in_h(LocalElementSize(n, n), TileElementSize(nb, nb));
   Matrix<T, Device::CPU> mat_out_h(LocalElementSize(n, n), TileElementSize(nb, nb));
 
-  dlaf::matrix::util::set(perms, [n](GlobalElementIndex i) { return n - 1 - i.row(); });
+  const matrix::Distribution& distr = mat_out_h.distribution();
+
+  SizeType index_start = distr.globalElementFromGlobalTileAndTileElement<C>(i_begin, 0);
+  SizeType index_finish = distr.globalElementFromGlobalTileAndTileElement<C>(i_end, 0) +
+                          distr.tileSize(GlobalTileIndex(i_end, i_end)).get<C>();
+  dlaf::matrix::util::set(perms, [index_start, index_finish](GlobalElementIndex i) {
+    if (index_start > i.row() || i.row() >= index_finish)
+      return SizeType(0);
+
+    return index_finish - 1 - i.row();
+  });
   dlaf::matrix::util::set(mat_in_h, [](GlobalElementIndex i) { return T(i.get<C>()); });
```

Comment:
The elements of the columns (resp. the rows depending on C) of the matrix have the same value.
I'm wondering if it would be better to choose a linear combination of row and col index.

Reply:
I'm worried about misalignment of rows/cols during the swap.
I suggest something like `return T(i.get<C>()) - T(i.get<OrthogonalToC>()) / 8;` which shouldn't overcomplicate the expected result.

Reply:
Is this OK? https://github.com/eth-cscs/DLA-Future/pull/564/commits/1c7ed5e5262c4a3ba957e5b5efd1b1a01da2c3d5